src/pages/settings/Wallet/UnshareBankAccount/UnshareBankAccount.tsx

```diff
@@ -0,0 +1,199 @@
+import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
+import useDebouncedState from '@hooks/useDebouncedState';
+import useLocalize from '@hooks/useLocalize';
+import useOnyx from '@hooks/useOnyx';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {formatMemberForList, getHeaderMessage, getSearchValueForPhoneOrEmail} from '@libs/OptionsListUtils';
+import {getPersonalDetailByEmail} from '@libs/PersonalDetailsUtils';
+import tokenizedSearch from '@libs/tokenizedSearch';
+import Navigation from '@navigation/Navigation';
+import type {PlatformStackScreenProps} from '@navigation/PlatformStackNavigation/types';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import {clearUnshareBankAccountErrors, unshareBankAccount} from '@userActions/BankAccounts';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+
+type ShareBankAccountProps = PlatformStackScreenProps<SettingsNavigatorParamList, typeof SCREENS.SETTINGS.WALLET.UNSHARE_BANK_ACCOUNT>;
+
+function UnshareBankAccount({route}: ShareBankAccountProps) {
+    const bankAccountID = route.params?.bankAccountID;
+    const styles = useThemeStyles();
+    const [bankAccountList] = useOnyx(ONYXKEYS.BANK_ACCOUNT_LIST, {canBeMissing: true});
+    const [showExpensifyCardErrorModal, setShowExpensifyCardErrorModal] = useState(false);
+    const currentUserPersonalDetails = useCurrentUserPersonalDetails();
+
+    const [countryCode = CONST.DEFAULT_COUNTRY_CODE] = useOnyx(ONYXKEYS.COUNTRY_CODE, {canBeMissing: false});
+
+    const [unsharedBankAccountData] = useOnyx(ONYXKEYS.UNSHARE_BANK_ACCOUNT, {canBeMissing: true});
+    const isLoading = unsharedBankAccountData?.isLoading ?? false;
+    const shouldShowSuccess = unsharedBankAccountData?.shouldShowSuccess ?? false;
+
+    const [unshareUser, setUnshareUser] = useState<{login?: string | null; text?: string | null} | undefined>(undefined);
+
+    const [searchTerm, debouncedSearchTerm, setSearchTerm] = useDebouncedState('');
+    const {translate} = useLocalize();
+    const admins = bankAccountList?.[bankAccountID]?.accountData?.sharees;
+    const totalAdmins = bankAccountList?.[bankAccountID]?.accountData?.sharees?.length;
+    const isExpensifyCardSettlementAccount = bankAccountList?.[bankAccountID]?.isExpensifyCardSettlementAccount ?? false;
+
+    const shouldShowTextInput = admins && admins?.length >= CONST.STANDARD_LIST_ITEM_LIMIT;
+    const textInputLabel = shouldShowTextInput ? translate('common.search') : undefined;
+
+    useEffect(() => {
+        if (!shouldShowSuccess) {
+            return;
+        }
+        if (!totalAdmins) {
+            Navigation.goBack();
+        }
+    }, [totalAdmins, shouldShowSuccess]);
+
+    const handleUnshare = () => {
+        if (!bankAccountID || !unshareUser?.login) {
+            return;
+        }
+        if (isExpensifyCardSettlementAccount) {
```

Comment:
A comment about why expensifycard settlement accounts require different behavior would be helpful here